sparkly/testing.py

```diff
@@ -120,28 +147,132 @@ def tearDown(self):
         for fixture in self.fixtures:
             fixture.teardown_data()
 
-    def assertDataFrameEqual(self, actual_df, expected_data, fields=None, ordered=False):
+    def assertDataFrameEqual(self,
+                             actual_df,
+                             expected_data,
+                             fields=None,
+                             ordered=False,
+                             places=None):
         """Ensure that DataFrame has the right data inside.
 
         Args:
             actual_df (pyspark.sql.DataFrame|list[pyspark.sql.Row]): Dataframe to test data in.
             expected_data (list[dict]): Expected dataframe rows defined as dicts.
             fields (list[str]): Compare only certain fields.
             ordered (bool): Does order of rows matter?
+            places (int|None): Compare floats up to given decimal places.
```

Comment:
Shi why to follow such confusing api's :) I really don't like it much more than your impl ;) , they looks to mimic some C lib api like matlab which is awful. Also does parameter name 'places' forces to look into docs only me? Am I able to test that deviation is not more than 0.2 for example?